Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -352,105 +373,84 @@ bool Digitizer::convertHits(const int det, const std::vector<HitType>& hits, TRD
         signalOld[1] = 0;
         signalOld[2] = 0;
         for (int iPad = 0; iPad < kNpad; iPad++) {
-          int colPos = colE + iPad - 1;
+          int colPos = pads_in_response[iPad];
           if (colPos < 0) {
             continue;
           }
           if (colPos >= nColMax) {
             break;
           }
           // Add the signals
-          // signalOld[iPad] = arraySignal.getData(rowE, colPos, iTimeBin); // TO BE ADDED LATER REQUIRES ARRAYSIGNALS
+          const int key = keys_in_response[iPad];
+          // Get the old signal
+          signalOld[iPad] = adcMapCont[key][iTimeBin];
           if (colPos != colE) {
             // Cross talk added to non-central pads
             signalOld[iPad] += padSignal[iPad] * (timeResponse + crossTalk);
           } else {
             // Without cross talk at central pad
             signalOld[iPad] += padSignal[iPad] * timeResponse;
           }
-          // arraySignal.setData(rowE, colPos, iTimeBin, signalOld[iPad]); // TO BE ADDED LATER REQUIRES ARRAYSIGNALS
-          // ******* TO BE ADDED LATER REQUIRES ARRAYSIGNALS
+          // Update the final signal
+          adcMapCont[key][iTimeBin] = signalOld[iPad];
         } // Loop: pads
       }   // Loop: time bins
     }     // end of loop over electrons
   }       // end of loop over hits
   return true;
 }
 
-bool Digitizer::convertSignalsToDigits(const int det, int& arraySignal)
+bool Digitizer::convertSignalsToDigits(const int det, ArrayADCMapContainer_t& adcMapCont)
 {
   //
   // Converstion of signals to digits
   //
 
   if (mSDigits) {
     // Convert the signal array to s-digits
-    if (!convertSignalsToSDigits(det, arraySignal)) {
+    if (!convertSignalsToSDigits(det, adcMapCont)) {
       return false;
     }
   } else {
     // Convert the signal array to digits
-    if (!convertSignalsToADC(det, arraySignal)) {
+    if (!convertSignalsToADC(det, adcMapCont)) {
       return false;
     }
     // Run digital processing for digits
     // RunDigitalProcessing(det);
   }
-  // Compress the arrays
-  // CompressOutputArrays(det);
   return true;
 }
 
-bool Digitizer::convertSignalsToSDigits(const int det, int& arraySignal)
+bool Digitizer::convertSignalsToSDigits(const int det, ArrayADCMapContainer_t& adcMapCont)
```

Comment:
I would to the conversion from signals to ADCs during the conversion from the map to the vector. Otherwise there will be two loops over the map.
Essentially, this would make the map a signal map, and the vector would contain digits.

Reply:
I agree, I will implement this an add it to the PR within this code review.

Reply:
I would keep the conversion separate by now, until we get all processes defined. Also, despite a bit of loss in performance, doing the conversion in a separate step makes it more readable.